internal/data/metric.go

```diff
@@ -24,62 +25,80 @@ import (
 // MetricData is the top-level struct that is propagated through the metrics pipeline.
 // This is the newer version of consumerdata.MetricsData, but uses more efficient
 // in-memory representation.
+//
+// Must use NewMetricData functions to create new instances.
+// Important: zero-initialized instance is not valid for use.
 type MetricData struct {
-	resourceMetrics []*ResourceMetrics
+	orig []*otlpmetric.ResourceMetrics
+
+	// Override a few fields. These fields are the source of truth. Their counterparts
+	// stored in corresponding fields of "orig" are ignored.
+	internal *internalMetricData
```

Comment:
Should use `impl` or `pimpl` instead of `internal` since this is the pimpl pattern?

Reply:
Done.